src/unix/poll.c

```diff
@@ -32,9 +32,8 @@ static void uv__poll_io(uv_loop_t* loop, uv__io_t* w, unsigned int events) {
   int pevents;
 
   handle = container_of(w, uv_poll_t, io_watcher);
-
-  if (events & POLLERR) {
-    uv__io_stop(loop, w, POLLIN | POLLOUT | UV__POLLRDHUP);
+  if ((events & POLLERR) && (events & UV__POLLPRI) != UV__POLLPRI) {
```

Comment:
Steps to reproduce (and you'll need that Raspberry Pi). On that Raspberry Pi:

```
1. clone and build libuv with my patches
2. inside the libuv folder clone wiringX
git clone --depth 1 -b rewrite https://github.com/wiringX/wiringX.git
3. build wiringX
cd wiringX
mkdir build
cd build
cmake ..
make
4. Go to the libuv root folder again
5. Use the source i posted here:
https://github.com/libuv/libuv/pull/1040#issuecomment-246054347
6. Compile with both libuv and wiringX
gcc -o poll-irq poll-irq.c .libs/libuv.a wiringX/build/libwiringX.a -IwiringX/src/ -lpthread -ldl
7. run poll-irq and set GPIO 1 to high or low by connecting and disconnecting to/from 3.3v.
```

If everything went well you should see the word `interrupt` show whenever you trigger an interrupt. This stops working when we solely have this line as you suggested:
`if (events & POLLERR) {`